src/worker/mod.rs

```diff
@@ -210,17 +208,9 @@ impl Worker {
         self.wf_task_source
             .wait_for_tasks_from_complete_to_drain()
             .await;
-        // Notify in case shutdown was requested while there were no more outstanding WFTs.
-        // This is required because the only other place where we notify wfts_drained is on
-        // activation completion and activation polling checks for wfts_drained.
-        self.maybe_notify_wtfs_drained();
         // wait until all outstanding workflow tasks have been completed before shutting down
-        let mut drained = self.wfts_drained.clone();
-        while !*drained.borrow() {
-            drained
-                .changed()
-                .await
-                .expect("wfts_drained should not be dropped");
+        while !self.all_wfts_drained() {
```

Comment:
Just verifying:
- Will there always be a WFT pending when we're handling LAs?
- Is there a way for us to get a new WFT from completion after this returns `true`?

Reply:
* Yes there needs to be
* Shouldn't be since `wait_for_tasks_from_complete_to_drain` can't finish until we're done getting such tasks